.github/workflows/CI.yml

```diff
@@ -3,6 +3,8 @@ name: Java CI with Gradle
 on:
   push:
     branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
Perhaps if we are going to run this build on PRs then we should add the following to the `Upload Applciation` step: 
```yml
if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

That way we only upload WAR files after something has been merged to main, and we can save space.

Reply:
Done